### DISCUSSION TITLE
Is it possible to publish static AAB files without applying any Android plugin?

### DISCUSSION THREAD

**Discussion by erawhctim** [Triple-T/gradle-play-publisher#1152] [Q&A] [answered]
I'm trying to build a very small Gradle file to essentially let me run this publishing plugin as a CLI script. 
I have no need to actually configure anything Android related (I have a pre-built AAB file to pass into Gradle). 

I tried build a Gradle file like this:
```
// build.gradle.kts
plugins {
  alias(libs.plugins.playPublisher)
}

play {
  serviceAccountCredentials = file("some-path-here")  
  defaultToAppBundles = true 
}
```

(`settings.gradle.kts` and libs TOML thing omitted, but hopefully you get the idea)

If I run it like this: 
```
./gradlew publishReleaseBundle \
    --artifact-dir build.aab \
    --release-status "draft" \
    --track "alpha"
```

I see this cryptic error:
```
* What went wrong:
com/android/build/api/AndroidPluginVersion
> com.android.build.api.AndroidPluginVersion
```

I'm assuming this might be due to the publishing plugin expecting the Android Gradle Plugin to be available/configured in my build 😕 Is this feasible or am I going down the wrong path with what I'm trying to accomplish?

**Comment by SUPERCILEX** [comment]
Not currently possible, but I'd accept a PR.

**Comment by erawhctim** [comment, marked as answer]
I dug into this a bit more and now I actually don't think a PR is really necessary: I was able to put together a minimal Android project for this without _too_ much work (or boilerplate); it works well! 

Copying here in case anyone else is interested: 
- Create a root `settings.gradle.kts` file like this:
  ```
  pluginManagement {
    repositories {
      gradlePluginPortal()
      mavenCentral()
      google()
    }
  }
  ```
- Create your `build.gradle.kts` file at the same root directory, like this: 
  ```
  plugins {
    alias(libs.plugins.agpApp)
    alias(libs.plugins.playPublisher)
  }

  repositories {
    mavenCentral()
    google()
  }

  android {
    compileSdk = 34
    namespace = ... // must match the package name on the AAB file
    defaultConfig {
      // these values don't matter, as your AAB file should dictate the version information
      versionCode = 1
      versionName = "1.0"
    }
  }

  play {
    serviceAccountCredentials = file(...) // static path to your AAB file
    defaultToAppBundles = true 
  }
  ```

then just call whatever publish command you want to use:
`./gradlew publishReleaseBundle --track "alpha" --artifact-dir ./build.aab --release-status "draft"`
